client/components/tinymce/plugins/contact-form/plugin.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External Dependencies
+ */
+import tinymce from 'tinymce/tinymce';
+import i18n from 'lib/mixins/i18n';
+import React from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import Gridicon from 'components/gridicon';
+import ContactFormDialog from './dialog';
+
+const contactForm = editor => {
+	let node;
+
+	editor.on( 'init', () => {
+		node = editor.getContainer().appendChild(
+			document.createElement( 'div' )
+		);
+	} );
+
+	editor.on( 'remove', () => {
+		React.unmountComponentAtNode( node );
+		node.parentNode.removeChild( node );
+		node = null;
+	} );
+
+	editor.addCommand( 'WP_ContactForm', () => {
+		React.render(
+			React.createElement( ContactFormDialog, {
+				visible: true,
+				editor: editor,
```

Comment:
Do you think `editor` will need to be a prop of the dialog? If you have access to it from within the component, I don't think you'd need to separately pass `onInsertMedia`. Personally I'd rather remove the `editor` prop, since you're not currently using it anyways, and it decouples the Dialog from TinyMCE.